Issue #4765 - Review GzipHandler inside ServletContextHandler.

sbordet opened the pull request:
Removed GzipHandler from ServletContextHandler.
Updated ServletContextHandlerTest.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

Closes #4765.

sbordet committed:
Issue #4765 - Review GzipHandler inside ServletContextHandler.

Removed GzipHandler from ServletContextHandler.
Updated ServletContextHandlerTest.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Issue #4765 - Review GzipHandler inside ServletContextHandler.

Fixed test failures.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

gregw commented on jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java:
```diff
@@ -652,16 +622,6 @@ public void setSecurityHandler(SecurityHandler securityHandler)
-    /**
-     * @param gzipHandler The {@link GzipHandler} to set on this context.
-     */
-    public void setGzipHandler(GzipHandler gzipHandler)
```
I think we should leave the get and set methods, but deprecate them.  The implementation can be replaced by an insert handler call

gregw commented on jetty-servlets/src/test/java/org/eclipse/jetty/servlets/IncludedGzipTest.java:
```diff
@@ -90,7 +90,7 @@ public void setUp() throws Exception
         GzipHandler gzipHandler = new GzipHandler();
         gzipHandler.setMinGzipSize(16);
-        tester.getContext().insertHandler(gzipHandler);
+        tester.getServer().insertHandler(gzipHandler);
```
This is not the equivalent test.  it should be `tester.getContext().insertHandler(gzipHandler);`

gregw replied:
perhaps test gzip in and outside of the context?

sbordet committed:
Issue #4765 - Review GzipHandler inside ServletContextHandler.

Fixed XML files.
Restored ServletContextHandler.setGzipHandler() and
deprecated it in case it's used from user's XML files.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Issue #4765 - Review GzipHandler inside ServletContextHandler.

Restored IncludedGzipTest.
Fixed ServletContextHandler.relinkHandlers(), now excluding
the checks for GzipHandler.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>
